apps/x509.c

```diff
@@ -591,6 +589,9 @@ int x509_main(int argc, char **argv)
         if (xca == NULL)
             goto end;
     }
+    issuer_cert = xca != NULL ? xca : x;
+    if (!X509_set_issuer_name(x, X509_get_subject_name(issuer_cert)))
+        goto end;
```

Comment:
Just move this inside the above if and you do not need to have the issuer_cert variable.